server/lib/nl/fulfillment/time_delta.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import List
+
+from lib.nl import utils
+from lib.nl.detection import ClassificationType
+from lib.nl.detection import Place
+from lib.nl.detection import RankingClassificationAttributes
+from lib.nl.detection import TimeDeltaClassificationAttributes
+from lib.nl.fulfillment import context
+from lib.nl.fulfillment.base import add_chart_to_utterance
+from lib.nl.fulfillment.base import ChartVars
+from lib.nl.fulfillment.base import overview_fallback
+from lib.nl.fulfillment.base import populate_charts
+from lib.nl.fulfillment.base import PopulateState
+from lib.nl.utterance import ChartOriginType
+from lib.nl.utterance import ChartType
+from lib.nl.utterance import Utterance
+
+
+#
+# Computes growth rate and ranks charts of comparable peer SVs.
+#
+def populate(uttr: Utterance):
+  time_delta_classification = context.classifications_of_type_from_utterance(
+      uttr, ClassificationType.TIME_DELTA)
+  logging.info('In Time-delta')
+
+  # Get time delta type
+  if (not time_delta_classification or
+      not isinstance(time_delta_classification[0].attributes,
+                     TimeDeltaClassificationAttributes) or
+      not time_delta_classification[0].attributes.time_delta_types):
+    return False
+  time_delta = time_delta_classification[0].attributes.time_delta_types
+
+  # Get potential ranking type
+  current_ranking_classification = context.classifications_of_type_from_utterance(
+      uttr, ClassificationType.RANKING)
+  ranking_types = []
+  if (current_ranking_classification and
```

Comment:
this if condition seems repeated in other places, consider a helper function? (or in later PR)

Reply:
SG for later PR.